### DISCUSSION TITLE
Help a Beginner: Best Integration of Sepia into openHAB 3 and Google Assistant

### DISCUSSION THREAD

**Discussion by SHU-red** [SEPIA-Framework/sepia-docs#177] [Q&A] [answered]
Right now i am using
- HomeServer > Docker > openHAB 3
- HomeServer > Docker > Sepia
- HomeServer > Docker > Sepia-STT
- myopenhab cloud service (connected to my openHAB 3)
- Google Assistant (connected to myopenhab cloud service)

I now want to try out Sepia and already got it running
At the moment i am about to connect to openHAB through Sepia

**But there are some questions for the long term:**
(i dont know by now)
1. Can i use my Google Home speakers? Directly?
2. Is a individual Hotword detection possible? How?
4. Is/Will there be a YouTube Music integration into Sepia? Maybe Google did not provide a interface? Thats why my SmartHome Speaker should also have a good audio quality!

Sorry if im Asking very basic questions and maybe i did not find/read/understand some of the documentation.
Id like to have a better view of where this journey could lead me!
Maybe someone has a good advice how to integrate Sepia into this environment especially trying to keep services Like YouTube Music in parrallel ...

Thanks

**Comment by fquirin** [comment, marked as answer]
> Can i use my Google Home speakers? Directly?

This I cannot really answer, it depends on the data you can read via Google Assistant + myopenhab. I'm assuming the GA has an openHAB skill? The path of one command would then be:
GA Home Device -> Google Cloud -> openHAB Cloud -> openHAB home server -> SEPIA home server ... 
instead of:
SEPIA client -> SEPIA home server :sweat_smile: 

> Is a individual Hotword detection possible? How?

Yes. SEPIA uses Porcupine and has support for a number of pre-defined words (v1.4-v1.9) and fully customizable words (v2+ with Porcupine access key). You can [read more here](https://github.com/SEPIA-Framework/sepia-html-client-app/blob/master/www/xtensions/picovoice/README.md)

> Is/Will there be a YouTube Music integration into Sepia?

There was a [discussion about this](https://github.com/SEPIA-Framework/sepia-docs/issues/138) (unfortunately in German) and the result was a little Python server I wrote that implements a hacky Python script as "unofficial YT-Music API" and works as a drop-in replacement for the default Youtube API (see SEPIA core settings). This is the server: https://github.com/fquirin/python-ytmusicapi-server 
It is a bit tricky to get everything running ...

**Reply by SHU-red** [reply]
>  instead of: SEPIA client -> SEPIA home server 😅

OK,
what kind of devices do you use as client?
Is there a hardware to pick to build a "SmartHome Speaker for Sepia"?
Or is all of this some RPi-Pico-DIY stuff?
Or do you always run a client on a smartphone/browser ...

> Yes. SEPIA uses Porcupine and has support for a number of pre-defined words (v1.4-v1.9) and fully customizable words (v2+ with Porcupine access key). You can [read more here](https://github.com/SEPIA-Framework/sepia-html-client-app/blob/master/www/xtensions/picovoice/README.md)

Awesome!

> There was a [discussion about this](https://github.com/SEPIA-Framework/sepia-docs/issues/138) (unfortunately in German) and the result was a little Python server I wrote that implements a hacky Python script as "unofficial YT-Music API" and works as a drop-in replacement for the default Youtube API (see SEPIA core settings). This is the server: https://github.com/fquirin/python-ytmusicapi-server It is a bit tricky to get everything running ...

Sounds good!
If im right this only works when using Sepia client in the browser?

**Reply by fquirin** [reply]
> what kind of devices do you use as client?
Is there a hardware to pick to build a "SmartHome Speaker for Sepia"?
Or is all of this some RPi-Pico-DIY stuff?
Or do you always run a client on a smartphone/browser ...

I'm using the Android app in combination with one [DIY Raspberry 4 client](https://github.com/SEPIA-Framework/sepia-installation-and-setup/tree/master/sepia-client-installation/rpi) in the living room (USB mic with Hyperpixel 4 touchscreen, [see lower right pic](https://twitter.com/sepia_fw/status/1498433293297111050/photo/1)) and one smaller RPi4 DIY where I constantly try new microphones (ReSpeaker Mic HATs, USB array etc.) and setups.
I hope I can offer some reference device at some point, but I still have to find the best components ^^.

> If im right this only works when using Sepia client in the browser?

Technically every client is the same (right now), but on iOS Safari I had some issues and Google is constantly messing with YouTube. I think the problem with YouTube Music was that you have to find a way to log in somehow and that is easier if you have full control over your browser tabs. If you use the DIY client you can achieve this via the remote access (there are some instructions in the DIY client setup menu).

**Reply by SHU-red** [reply]
Nice devices you have there!
Really really sounds interesting ...

Im very curious to step more and more into sepia and maybe get something going parallel to my current setup.
I will be trying out more and more, as far as my time allows.

**Reply by fquirin** [reply]
Happy you like it :-)
The next update is around the corner, just have to do some final testing and wrap everything up, so keep an eye out for new releases ;-)

**Reply by SHU-red** [reply]
Docker Watchtower is watching you XD

**Reply by SHU-red** [reply]
> There was a [discussion about this](https://github.com/SEPIA-Framework/sepia-docs/issues/138) (unfortunately in German) and the result was a little Python server I wrote that implements a hacky Python script as "unofficial YT-Music API" and works as a drop-in replacement for the default Youtube API (see SEPIA core settings). This is the server: https://github.com/fquirin/python-ytmusicapi-server
> It is a bit tricky to get everything running ...

One more question about this [YT Music MicroServer](https://github.com/fquirin/python-ytmusicapi-server)
1. I would try to create my own docker container for this server (which will be very hard because im absolutely not advanced in this but ill try)
2. What will i have to do / how is it possible to use this YTM-server through sepia? Im not seeing any YTM-Specific settings. (Im having a YT-Premium account; Where to enter the adress/port of this service?)

**Comment by SHU-red** [comment]
I have no idea how i could get this working so fast.
But im pretty sure, this is because of the awesome documentaion-work done by @fquirin 

Here i created a dockerized version of your YTM microservice which seems to be working in my case:
https://hub.docker.com/repository/docker/shur3d/ytmusicapi_microservice

**Reply by fquirin** [reply]
I think the public URL (without Docker login) is: https://hub.docker.com/r/shur3d/ytmusicapi_microservice :slightly_smiling_face: 
Just a small warning: Google can break this anytime and probably will as YT Music keeps evolving ... we will see :grimacing:

**Reply by SHU-red** [reply]
Thanks you already mentioned that
I'm just very glad that I had the chance (and maybe ability) to contribute somehow and not only open issues and ask questions 😅

**Reply by fquirin** [reply]
well any feedback helps to find bugs and weak spots in the documentation and makes it hopefully easier for the next user :wink: :+1:
